### PR TITLE
Adding optional IAM policy creation for EKS Console

### LINKED ISSUE #2189: Optional IAM policy creation for EKS Console as an option in clusterawsadm bootstrap
/kind feature
/area provider/eks
/help
/good-first-issue

**Describe the solution you'd like**
It would be good to have an option to create a IAM policy for use with the EKS console (based on this [example](https://docs.aws.amazon.com/eks/latest/userguide/security_iam_id-based-policy-examples.html#policy_example3)) as part of:  `clusterawsadm bootstrap iam create-cloudformation-stack --config eks.config`. The option would need to be explicitly enabled via a supplied config file. 

**Environment:**

- Cluster-api-provider-aws version: 0.6.3
- Kubernetes version: (use `kubectl version`):  1.18

### PR DESCRIPTION
**Pull request by Madhur97**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adding an option to create a new IAM policy with the required permissions to view nodes/workloads using the EKS console so that this policy can be assigned to users/groups as mentioned [here](https://cluster-api-aws.sigs.k8s.io/topics/eks/eks-console.html)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2189 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/clusterawsadm/api/bootstrap/v1alpha1/types.go [hunk 1]
```diff
@@ -96,6 +96,8 @@ type EKSConfig struct {
 	// AllowIAMRoleCreation controls whether the EKS controllers have permissions for creating IAM
 	// roles per cluster
 	AllowIAMRoleCreation bool `json:"iamRoleCreation,omitempty"`
+	// EnableEKSViewNodesAndWorkloadsPolicy controls the creation of the policy to view EKS nodes and workloads.
+	EnableEKSViewNodesAndWorkloadsPolicy bool `json:"enableEKSViewNodesAndWorkloadsPolicy"`
```
Comment (sedefsavas) [inline cmd/clusterawsadm/api/bootstrap/v1alpha1/types.go:100]: Too long?
```suggestion
	EnableNodesAndWorkloadsPolicy bool `json:"enableEKSViewNodesAndWorkloadsPolicy"`
```

Also is this omitempty?

Comment (Madhur97) [reply cmd/clusterawsadm/api/bootstrap/v1alpha1/types.go]: Updated the name and added omitempty

File: cmd/clusterawsadm/cloudformation/bootstrap/template.go [hunk 2]
```diff
@@ -194,6 +195,14 @@ func (t Template) RenderCloudFormation() *cloudformation.Template {
 		}
 	}
 
+	if t.Spec.EKS.EnableNodesAndWorkloadsPolicy {
+		template.Resources[string(EKSViewNodesAndWorkloadsPolicy)] = &cfn_iam.ManagedPolicy{
+			ManagedPolicyName: t.NewManagedName("eks-view-nodes-workloads"),
```
Comment (richardcase) [inline cmd/clusterawsadm/cloudformation/bootstrap/template.go:200]: I think we should shorten the name of this as its going to have a suffix appended and optionally a prefix which will make it long.

Perhaps something like `eks-console`?

**Review by richardcase — CHANGES_REQUESTED** [review]

**Comment by sedefsavas** [comment]
@Madhur97 
what is the status on this PR?

**Comment by Madhur97** [comment]
Need to close this PR due to some issues. Opened a new [PR](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2790) .

**Pull request closed without merging**